Request ProjectOpen with Number(3) failed due to ProjectNotFound

Every **first** time I try to create project by `enso --new` and then try to open it by project manager, I get an error:
```
[INFO] [2023-12-12T06:57:36+01:00] [org.enso.projectmanager.boot.ProjectManager$] Started server at 127.0.0.1:30535, press enter to kill server
[INFO] [2023-12-12T07:00:50+01:00] [org.enso.projectmanager.protocol.ClientController] Client connected to Project Manager [964a1ec9-7f7f-40ff-abe3-f01b7613167b]nso / runProjectManagerDistribution 193s
[ERROR] [2023-12-12T07:00:52+01:00] [org.enso.projectmanager.requesthandler.ProjectOpenHandler] Request ProjectOpen with Number(3) failed due to ProjectNotFound
```

When I stop the `project-manager` and start it again, everything **works for the second time**!

The exact steps to reproduce I use every [AoC day](https://adventofcode.com/2023):
```bash
sbt:enso> runEngineDistribution --new /home/devel/enso/projects/Dec_12
sbt:enso> runProjectManagerDistribution
```
and then from another terminal
```bash
enso$ ./run gui watch --skip-wasm-opt
```
then I connect to http://localhost:8080 via chrome browser and the above described failure happens. After the second (successful) open I see following files changed on disk:
```
~/enso/projects/Dec_12$ git status
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   src/Main.enso

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .enso/
```
The `src/Main.enso` newly contains metadata (added by the IDE).

--- a/lib/scala/project-manager/src/main/scala/org/enso/projectmanager/service/ProjectService.scala
+++ b/lib/scala/project-manager/src/main/scala/org/enso/projectmanager/service/ProjectService.scala
@@ -10,7 +10,7 @@ import org.enso.pkg.validation.NameValidation
 import org.enso.projectmanager.control.core.syntax._
 import org.enso.projectmanager.control.core.CovariantFlatMap
 import org.enso.projectmanager.control.effect.syntax._
-import org.enso.projectmanager.control.effect.{ErrorChannel, Sync}
+import org.enso.projectmanager.control.effect.{ErrorChannel, Semaphore, Sync}
 import org.enso.projectmanager.data.{
   LanguageServerStatus,
   MissingComponentAction,
@@ -70,6 +70,8 @@ class ProjectService[
     extends ProjectServiceApi[F] {
 
   private lazy val logger = Logger[ProjectService[F]]
+
+  private val listProjectsLock: Semaphore[F] = Semaphore.unsafeMake(1)
 
   import E._
 
@@ -359,8 +361,8 @@ class ProjectService[
   override def listProjects(
     maybeSize: Option[Int]
   ): F[ProjectServiceFailure, List[ProjectMetadata]] =
-    repo
-      .getAll()
+    listProjectsLock
+      .withPermit(repo.getAll())
       .map(
         _.sorted(RecentlyUsedProjectsOrdering)
           .take(maybeSize.getOrElse(Int.MaxValue))

